Sources/Commands/SwiftTestTool.swift

```diff
@@ -288,6 +292,38 @@ public class SwiftTestTool: SwiftTool<TestToolOptions> {
                 executionStatus = .failure
             }
         }
+
+        if options.shouldEnableCodeCoverage {
+            let buildParameters = try self.buildParameters()
+            let toolChain = try getToolchain()
+            try mergeRawDataFiles(fs: fileManager, path: buildParameters.buildPath.appending(component: "codecov"), toolChain: toolChain)
+            let graph = try loadPackageGraph()
+            for packages in graph.rootPackages {
+                let component = packages.name + "PackageTests"
+                let testPath = try buildTestsIfNeeded(options, graph: graph).appending(component: "Contents").appending(component: "MacOS").appending(components: component)
+                try generateReport(fs: fileManager, testPath: testPath, buildPath: buildParameters.buildPath, name: packages.name, toolChain: toolChain)
+            }
+        }
+    }
+
+    private func mergeRawDataFiles(fs: FileSystem, path: AbsolutePath, toolChain: UserToolchain) throws {
+            let files = try fs.getDirectoryContents(path)
+            let llvmProf = try toolChain.getLlvmProf()
+            var args: [String] = [llvmProf.asString, "merge", "-sparse"]
+            for file in files {
+                args.append(path.appending(component: file).asString)
+            }
+            args += ["-o",  path.appending(component: "default.profdata").asString]
+            try Process.checkNonZeroExit(arguments: args)
+    }
+
+    private func generateReport(fs: FileSystem, testPath: AbsolutePath, buildPath: AbsolutePath, name: String, toolChain: UserToolchain) throws {
+        let codeCoveragePath = buildPath.appending(component: "codecov").appending(component: "default.profdata")
+        let llvmCov = try toolChain.getLlvmCov()
+        let args: [String] = [llvmCov.asString, "export", "-instr-profile="+codeCoveragePath.asString, testPath.asString]
```

Comment:
I think we don't need the explicit type here.

Reply:
done